NewForce-at-Mountwest/bangazonapi-team-crocodoom
# PR for Product Type Controller and Test Controller

## Description
This is a build for the Product Type Controller, as well as the Product Type Controller. The Product Type Controller allows the Developer to use the BangazoneAPI to GET all Product Types, GET a single Product Type, Create (POST) a new Product type to the Database, Update (PUT) a single Product Type, and Soft DELETE a single Product Type. Using the BangazonAPITest, the Product Type Test Controller allows the Developer to test all of the previously mentioned methods with dummy data. 
(The Branch name is currently perryPaymentType. This is a mistake and should be for Product Type.)

## Proposed changes
This is a new feature that adds functionality to Product Type
- New Models were added for Product Type as well as Controllers for Product Type and Testing
## Related Issues
- A possible issue could be a database Conflict
## Steps To Test
Before Testing, rebuild your database by running the SQL command listed in bangazon-api.sql. This should wipe the Database while also seeding it, making it consistent to the Database needed to run these Methods.

You will need to git checkout into perryPaymentType branch. Cd into the BangazonAPI and use "dotnet run" to run the Program. Then open up a window of Postman in order to test each individual method. The url should be http://localhost:63540/api/ProductType to view all Product Types.

1.- GET - The first method to test is GET. This will get all of the product types in the database. Copy the Url in the web browser and paste it in postman. Run the GET command and all of the Product Types should be shown.

2.- GET- The second method to test is GET to find a single Product Type by id. Using the url from step 1, add a "/1" to view the product type with the Primary Key(id) of 1. Run the GET command and the Product Type should be shown.

3.- POST - The third method to test is POST. This will allow the Developer to POST a new Product Type to the Database. Using the original url in step 1, change the method POST int postman. Add a Product Type to the database by adding a Name and the boolean IsActive. Use the GET method to view the list of all Product Types including the one just created.

4.- PUT - The next method to test is PUT. This will allow the Developer to update an individual Product Type in the Database. Using the url from step two, change the method in Postman to PUT and change the Name to "Test". Use the GET method to view the updated Product Type.

5.- DELETE - The last method to test is DELETE. This will allow the Developer to soft delete an individual Product Type. This will not actually Hard Delete a Product Type, rather just change the IsActive boolean to false. Using the url from steps 2 and 4, change the method in Postman to DELETE and soft Delete the current Product type. Use the GET method to view the updated Product Type with an updated IsActive boolean.

6.- Go to View in the toolbar of Visual Studio, select Test Explorer. We are going to test the Methods by creating dummy data in the productTypeTestController. In the top left of the Test explorer, click the green play button to run the Testing tasks. The results for each task should be a green arrow.
## What kind of change does this PR introduce?
Please check one of the following:
- [X ] New feature
